cluster/cluster.go

```diff
@@ -13,6 +13,7 @@ type Lock interface {
 	// through this interface. A context is accepted for setting wait bounds.
 	Lock(context.Context) error
 	Unlock(context.Context) error
+	UnlockLogError(context.Context)
```

Comment:
Still going through, but are you planning on using this everywhere? Better to keep the interface simple, and just remove the error `Unlock` returns? Doesn't feel super go like I guess, but gets the functionality you're looking for.

Reply:
I debated that one; the main plus for adding it was that it would be a very common pattern to call `Unlock` in a `defer` statement (which the Register does in a later commit in this PR); this intended to mimic the familiar semantics of `sync.Mutex.Lock()` followed with a `defer sync.Mutex.Unlock()`.

One alternative is to defer a wrapper func inside your code:

```go
func someFuncThatLocks() {
   ...
   l.Lock(ctx)
   defer unlockLogError(l, ctx)
   ...
}

func unlockLogError(l ZooKeeperLock, ctx context.Context) {
  if err := l.Unlock(ctx); err != nil {
    log.Println(err)
  }
}
```